web3c/deploy_header.js

```diff
@@ -0,0 +1,247 @@
+   */
+  static prefix() {
+    return '00736973';
+  }
+}
+
+/**
+ * A collection of utilities for parsing through deploycode including the Oasis contract
+ * deploy header in the form of a hex string.
+ */
+class DeployHeaderHexReader {
+  /**
+   * @param   {String} deploycode is the transaction data to deploy a contract as a hex string.
+   * @returns the contract deploy header prefixed to the deploycode, otherwise, null.
+   */
+  static header(deploycode) {
+    if (!deploycode.startsWith('0x' + DeployHeader.prefix())) {
+      return null;
+    }
+    let version = DeployHeaderHexReader.version(deploycode);
+    let body = DeployHeaderHexReader.body(deploycode);
+
+    if (!DeployHeader.isValidBody(body)) {
+      throw Error(`Invalid body ${body}`);
+    }
+
+    return new DeployHeader(version, body);
+  }
+  /**
+   * @param {String} deploycode is a hex string of the header || initcode.
+   */
+  static body(deploycode) {
+    assert.equal(true, deploycode.startsWith('0x' + DeployHeader.prefix()));
+
+    let length = DeployHeaderHexReader.size(deploycode);
+    let serializedBody = deploycode.substr(DeployHeaderHexReader.bodyStart(), length*2);
+
+    return JSON.parse(
+      Buffer.from(serializedBody, 'hex').toString('utf8')
+    );
+  }
+
+  /**
+   * @param {String} deploycode is a hex string of the header || initcode.
+   */
+  static size(deploycode)  {
+    assert.equal(true, deploycode.startsWith('0x' + DeployHeader.prefix()));
+
+    let length = deploycode.substr(
+      DeployHeaderHexReader.sizeStart(),
+      DeployHeaderHexReader.sizeLength()
+    );
+
+    return parseInt('0x' + length);
+  }
+
+  /**
+   * @param {String} deploycode is a hex string of the header || initcode.
+   */
+  static version(deploycode) {
+    assert.equal(true, deploycode.startsWith('0x' + DeployHeader.prefix()));
+
+    let version = deploycode.substr(
+      DeployHeaderHexReader.versionStart(),
+      DeployHeaderHexReader.versionLength()
+    );
+
+    return parseInt('0x' + version);
+  }
+
+  /**
+   * @param {String} deploycode is a hex string of the header || initcode.
+   */
+  static initcode(deploycode) {
+    assert.equal(true, deploycode.startsWith('0x' + DeployHeader.prefix()));
+
+    return '0x' + deploycode.substr(
+      DeployHeaderHexReader.initcodeStart(deploycode)
+    );
+  }
+
+  static initcodeStart(deploycode) {
+    assert.equal(true, deploycode.startsWith('0x' + DeployHeader.prefix()));
+
+    // Make sure to convert the "length" to nibbles, since it's in units of bytes.
+    return DeployHeaderHexReader.bodyStart() + DeployHeaderHexReader.size(deploycode)*2;
+  }
+
+  /**
+   * @returns the hex string index of the start section.
+   */
+  static versionStart() {
+    return 2 + DeployHeader.prefix().length;
+  }
+
+  /**
+   * @returns the length of the version in nibbles.
+   */
+  static versionLength() {
+    return 2*2;
+  }
+
+  /**
+   * @returns the length, in nibbles, of the size section.
```

Comment:
the index where the size section starts?

Reply:
Done.